pkg/apis/cluster/common/consts.go

```diff
@@ -103,3 +103,59 @@ const (
 	// i.e gradually scale down the old MachineSet and scale up the new one.
 	RollingUpdateMachineDeploymentStrategyType MachineDeploymentStrategyType = "RollingUpdate"
 )
+
+// MachinePhase is the ongoing-phase machine is going through at the moment.
+type MachinePhase string
+
+// These are the valid values for the MachinePhase.
+const (
+	// MachinePending should be set when machine is being created but it has not joined the
+	// cluster yet.
+	MachinePending MachinePhase = "Pending"
+
+	// MachineRunning should be set when machine has joined the cluster and running successfully.
+	MachineRunning MachinePhase = "Running"
+
+	// MachineTerminating should be set when machine is being deleted.
+	MachineTerminating MachinePhase = "Terminating"
+
+	// MachineUnknown should be set when there is no way for controller to understand the
+	// current phase of the machine. For example when is not reachable due to network-issues.
+	MachineUnknown MachinePhase = "Unknown"
+
+	// MachineFailed should be set when machine has failed and certain action needs to be taken either by higher-level controller or an admin.
+	MachineFailed MachinePhase = "Failed"
```

Comment:
Why not simply use `.Status.ErrorReason` and `.Status.ErrorMessage` for this when targeting humans?

Other controllers like the `MachineSet` controller already use [`.Status.NodeRef`]( https://github.com/kubernetes-sigs/cluster-api/blob/fee897706a82938117591515fbd4e85e667299a7/pkg/controller/machineset/status.go#L55-L60) and IMHO should continue to do so as [node health checking is a tricky problem we should not re-implement](https://github.com/kubernetes/community/blob/master/keps/sig-node/0009-node-heartbeat.md)